tools/sqlite3_api_wrapper/test/include/udf_scalar_functions.hpp

```diff
@@ -196,3 +210,23 @@ static void sum_overload_function(sqlite3_context *context, int argc, sqlite3_va
 
 	sqlite3_result_int(context, value1 + value2 + value3);
 }
+
+// calling sqlite3_value_text() multiple times
+static void calling_value_text_multiple_times(sqlite3_context *context, int argc, sqlite3_value **argv) {
+	assert(sqlite3_value_type(argv[0]) == SQLITE_INTEGER || sqlite3_value_type(argv[0]) == SQLITE_FLOAT ||
+	       sqlite3_value_type(argv[0]) == SQLITE_TEXT);
+	char *str = (char *)sqlite3_value_text(argv[0]);
+	assert(sqlite3_value_type(argv[0]) == SQLITE_TEXT);
+	auto len = strlen(str);
+	size_t len2;
+	char *str2;
+	// calling sqlite3_value_text multiple times,  i.e., 10x
+	for (size_t i = 0; i < 10; ++i) {
+		str2 = (char *)sqlite3_value_text(argv[0]);
+		assert(str == str2);
```

Comment:
Are these functions only used in tests? If so, the asserts should be converted into `REQUIRE`.  I think this is also where the unused variable warning comes from - `assert` is removed in release mode, and after the asserts are removed `len2` is unused. `REQUIRE` will not be compiled out.